projects/plugins/jetpack/extensions/blocks/blogging-prompt/edit.js

```diff
@@ -1,42 +1,143 @@
-import { BlockIcon } from '@wordpress/block-editor';
-import { Placeholder } from '@wordpress/components';
-// import { useState } from '@wordpress/element';
-import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import { InspectorControls, useBlockProps } from '@wordpress/block-editor';
+import { PanelBody, Spinner, ToggleControl, withNotices } from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+import { __, _n, _x, sprintf } from '@wordpress/i18n';
 import './editor.scss';
 import icon from './icon';
 
-function BloggingPromptsBetaEdit( {
-	// attributes,
-	className,
-	// noticeOperations,
-	noticeUI,
-	// setAttributes,
-} ) {
-	/**
-	 * Write the block editor UI.
-	 *
-	 * @returns {object} The UI displayed when user edits this block.
-	 */
-	// const [ notice, setNotice ] = useState();
-
-	/* Call this function when you want to show an error in the placeholder. */
-	// const setErrorNotice = () => {
-	// 	noticeOperations.removeAllNotices();
-	// 	noticeOperations.createErrorNotice( __( 'Put error message here.', 'jetpack' ) );
-	// };
+function BloggingPromptsBetaEdit( { attributes, noticeOperations, noticeUI, setAttributes } ) {
+	const [ isLoading, setLoading ] = useState( true );
+	const { answerCount, gravatars, prompt, promptId, showLabel, showResponses } = attributes;
```

Comment:
where does promptId get set? it comes through as todays prompt id, but not from the `answer_prompt` query string as I'd expect

Reply:
For a block that's newly inserted, the promptId will be empty be default, and then filled in with the most recent prompt for today. For an existing block, the promptId will be stored, and fresh data pulled in case information about prompt answers has changed.

I'm not handling the integration with the existing `blogging-prompts` extension yet, which uses the answer_prompt param  (not to be confused with this `blogging-prompt` beta extension being used to develop this new block).

To keep the PRs more manageable, I'm developing the block on it's own. When we're happy with it, I'll merge the two extensions, use the answer_prompt param to insert a writing prompt block with the promptId pre-filled, and mark as experimental, to release to WP.com sites.

Reply:
ok, that makes sense 👍